### PR TITLE
SQL Tuning: add the page `blocklist-control-plan.md`

### PR DESCRIPTION
**Pull request by winoros**
### What is changed, added or deleted? (Required)

Translating https://github.com/pingcap/docs-cn/pull/3253

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3253
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/3970

### CODE REVIEW

**Comment by yikeke** [comment]
The docs-cn PR https://github.com/pingcap/docs-cn/pull/3253 was also cherry-picked to the 3.0 and 3.1 branches. And we
need to align this behavior for this PR? @winoros

**Comment by winoros** [comment]
@yikeke ok, i updated the status.

**Comment by wwar** [comment]
May I suggest renaming from `blacklist` to `block` or `deny` list?  See: https://github.com/rails/rails/issues/33677 and https://go-review.googlesource.com/c/go/+/236857/ 

I realize it might be hard to rename the table `mysql.opt_rule_blacklist` without causing incompatibility, but the documentation could be a little bit more sensitive.

**Comment by ilovesoup** [comment]
> May I suggest renaming from `blacklist` to `block` or `deny` list? See: [rails/rails#33677](https://github.com/rails/rails/issues/33677) and https://go-review.googlesource.com/c/go/+/236857/
> 
> I realize it might be hard to rename the table `mysql.opt_rule_blacklist` without causing incompatibility, but the documentation could be a little bit more sensitive.

I agree that we can change document description to blocklist but leave keyword aside for compatibility.

File: TOC.md [hunk 1]
```diff
@@ -106,7 +106,7 @@
         + [Optimizer Hints](/optimizer-hints.md)
         + [SQL Plan Management](/sql-plan-management.md)
         + [Access Tables Using `IndexMerge`](/index-merge.md)
-        + [The Blacklist of Optimization Rules and Expression Pushdown](/blacklist-control-plan.md)
+        + [The Blacklist of Optimization Rules and Expression Pushdown](/blocklist-control-plan.md)
```
Comment (SunRunAway) [inline TOC.md:109]:
```suggestion
        + [The Blocklist of Optimization Rules and Expression Pushdown](/blocklist-control-plan.md)
```

File: blocklist-control-plan.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    | id                      | estRows  | task      | access object | operator info                      |
+    +-------------------------+----------+-----------+---------------+------------------------------------+
+    | Selection_7             | 10000.00 | root      |               | gt(ssb_1.t.a, 2), lt(ssb_1.t.a, 2) |
+    | └─TableReader_6         | 10000.00 | root      |               | data:TableFullScan_5               |
+    |   └─TableFullScan_5     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo     |
+    +-------------------------+----------+-----------+---------------+------------------------------------+
+    3 rows in set (0.00 sec)
+    ```
+
+4. Remove one expression(here is `>`) from the blacklist and execute `admin reload expr_pushdown_blacklist`.
```
Comment (SunRunAway) [inline blocklist-control-plan.md:204]:
```suggestion
4. Remove one expression(here is `>`) from the blocklist and execute `admin reload expr_pushdown_blacklist`.
```

File: blocklist-control-plan.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+title: The Blocklist of Optimization Rules and Expression Pushdown
+summary: Learn about the blocklist to control the optimization rules and the behavior of expression pushdown.
+category: performance
+---
+
+# The Blocklist of Optimization Rules and Expression Pushdown
+
+This document introduces how we use the blocklist of optimization rules and the blocklist of expression pushdown to control the behavior of TiDB.
+
+## The Blocklist of Optimization Rules
```
Comment (lilin90) [inline blocklist-control-plan.md:11]: We use the sentence-style capitalization for headings in the body.
```suggestion
## The blocklist of optimization rules
```

File: blocklist-control-plan.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+### Important Optimization Rules
```
Comment (lilin90) [inline blocklist-control-plan.md:13]:
```suggestion
### Important optimization rules
```

File: blocklist-control-plan.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+| Projection Elimination | projection_eliminate | Remove unnecessary projects |
+| Max/Min Elimination | max_min_eliminate | Rewrite some max/min function in aggregation to the form `order by` + `limit 1` |
+| Predicate Pushdown | predicate_push_down | Try to push predicates to the executor that is closer to the data source |
+| Outer Join Elimination | outer_join_eliminate | Try to convert the left/right join to inner join |
+| Partition Pruning | partition_processor | Pruning partitions which are rejected by the predicates and rewrite to the form `UnionAll + Partition Datasource` |
+| Aggregation Pushdown | aggregation_push_down| Try to push aggregations down to their children |
+| TopN Pushdown | topn_push_down | Try to push the topn to where is closer to the data source |
+| Join Reorder | join_reorder | Decide the order of multi-table joins |
+
+### Disable Optimization Rules
```
Comment (lilin90) [inline blocklist-control-plan.md:29]:
```suggestion
### Disable optimization rules
```

File: blocklist-control-plan.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+- If you want to re-enable some rules, delete the corresponding data in the table, and then run `admin reload` statement:
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    delete from mysql.opt_rule_blacklist where name in ("join_reoder", "topn_push_down");
+    admin reload opt_rule_blacklist;
+    ```
+
+## The Blocklist of Expression Pushdown
```
Comment (lilin90) [inline blocklist-control-plan.md:68]:
```suggestion
## The blocklist of expression pushdown
```

File: blocklist-control-plan.md [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+
+1. Delete the corresponding function name or operator name and the set of components you want to disable the pushdown from the table `mysql.expr_pushdown_blacklist`.
+
+2. Execute `admin reload expr_pushdown_blacklist`.
+
+> **Note:**
+>
+> `admin reload expr_pushdown_blacklist` only takes effect on the TiDB Server which runs this statement. If you need all TiDB Server of the cluster to disable some rules, you need to run this command on each of the TiDB Server.
+
+## Example of the Expression Blocklist
```
Comment (lilin90) [inline blocklist-control-plan.md:137]:
```suggestion
## Expression blocklist usage example
```

**Comment by lilin90** [comment]
@winoros Please take a look at my update in https://github.com/pingcap/docs/pull/3043/commits/b0e1ffde42164a803ebb21c4343485ccdec4411b.
@lzmhhh123 @SunRunAway PTAL again~ Thanks!

File: blocklist-control-plan.md [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+    > **Note:**
+    >
+    > `admin reload opt_rule_blacklist` only takes effect on the TiDB server where the above statement has been run. If you want all TiDB servers of the cluster to take effect, run this command on each TiDB server.
+
+- If you want to re-enable a rule, delete the corresponding data in the table, and then run the `admin reload` statement:
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    delete from mysql.opt_rule_blacklist where name in ("join_reoder", "topn_push_down");
```
Comment (lilin90) [inline blocklist-control-plan.md:65]:
```suggestion
    delete from mysql.opt_rule_blacklist where name in ("join_reorder", "topn_push_down");
```

**Comment by lilin90** [comment]
> Only `>` is re-pushed again.

@winoros Do you mean [the Chinese version](https://docs.pingcap.com/zh/tidb/dev/blacklist-control-plan#%E8%A1%A8%E8%BE%BE%E5%BC%8F%E9%BB%91%E5%90%8D%E5%8D%95%E7%94%A8%E6%B3%95%E7%A4%BA%E4%BE%8B) is incorrect? `重新观察执行计划，< 和 > 表达式又可以重新被下推到 TiKV Coprocessor。` Could you please make it more clearly?

File: blocklist-control-plan.md [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+
+    ```sql
+    admin reload expr_pushdown_blacklist;
+    ```
+
+    ```sql
+    Query OK, 0 rows affected (0.00 sec)
+    ```
+
+5. Observe the execution plan again and you will find that the `<` and `>` operators can be re-pushed down to TiKV Coprocessor again.
```
Comment (lilin90) [inline blocklist-control-plan.md:234]: @lzmhhh123 I've confirmed with @winoros that the Chinese meaning should be `重新观察执行计划，可以看到只有 > 表达式被重新下推到 TiKV Coprocessor，< 表达式仍然被禁用下推` and I have updated the Chinese version in https://github.com/pingcap/docs-cn/pull/3970.
```suggestion
5. Observe the execution plan again and you will find that `<` is not pushed down while `>` is pushed down to TiKV Coprocessor.
```

**Comment by yikeke** [comment]
LGTM

**Pull request merged**